File: Data Structures/Array/1 D/Micro and Array Update.cpp
//https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/micro-and-array-update/
#include<iostream>

using namespace std;

int main()
    {
        int t,k,i,j;
        long n,a[1000000],small=9999999;
        cin>>t;
        while(t--)
            {
                cin>>n>>k;
                small=9999999;
                for(i=0;i<n;i++)
                {
                    cin>>a[i];
                    if(a[i]<small)
                        small=a[i];
                }
                j=k-small;
                if(j>0)
                    cout<<j<<endl;
                else
                    cout<<"0"<<endl;
            }
        
        return 0;
    }

File: Data Structures/Array/1 D/Micro and Array Update.py
#https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/micro-and-array-update/
t = int(input())
while t:
    t-=1
    a,b = input().split()
    n=int(a)
    k=int(b)
    numbers = [int(n) for n in input().split()]
    small = 0
    for i in range(n):
        if i == 0:
            small=numbers[i]
        else:
            if small > numbers[i]:
                small = numbers[i]
    
    print(k-small)